### PR TITLE
Integrate gutenberg-mobile release 1.35.0

### PR DESCRIPTION
**Pull request by ceyhun**
## Description
This PR incorporates the 1.35.0 release of gutenberg-mobile.  
For more information about this release and testing instructions, please see the related Gutenberg-Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2557

Release Submission Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by geriux** [comment]
## Sanity testing

DarkMode-2

- [ ] Spacer block - Dark mode - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/darkmode.md#tc004)
- [ ] Buttons block - Dark mode - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/darkmode.md#tc005)
- [ ] Group - Dark mode - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/darkmode.md#tc007)
- [ ] Columns - Dark mode - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/darkmode.md#tc007)

Group-2

- [ ] Group - Navigation up button works as expected - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc009)
- [ ] Group - Navigation down works according to parent-first approach - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc010)
- [ ] Group - Cross navigation between blocks works as expected - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc011)
- [ ] Group - Ungroup button works as expected - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc012)
- [ ] Group - Check if in DarkMode all components gets proper colors - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc013)
- [ ] Group - Check if nested Placeholder block can be replaced - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc014)

Spacer-2

- [ ] Spacer block - Settings: Available height range is correct - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/spacer.md#tc004)
- [ ] Spacer block - Settings: Height range extends if Spacer comes from the web is higher than 500px - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/spacer.md#tc005)
- [ ] Spacer block - Spacer in horizontal layout works as expected - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/spacer.md#tc006)

Buttons-4

- [ ] Buttons block - Render custom text color - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/buttons.md#tc017)
- [ ] Buttons block - Render custom background color - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/buttons.md#tc016)
- [ ] Buttons block - Render gradient background color - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/buttons.md#tc018)
- [ ] Buttons block - Check if selection / caret color matches font color - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/buttons.md#tc020)
- [ ] Buttons block - The newly created Buttons has set background color to the theme-default color in preview - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/buttons.md#tc021)

Button-5

- [ ] Buttons block - Edit text styles - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/buttons.md#tc009)
- [ ] Buttons block - Render theme colors - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/buttons.md#tc019)
- [ ] Buttons block - Settings: Change Button border radius - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/buttons.md#tc015)

Button-6

- [ ] Buttons block - Settings: Open in new tab - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/buttons.md#tc011)
- [ ] Buttons block - Settings: Link rel - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/buttons.md#tc012)
- [ ] Buttons block - Settings: Link URL - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/buttons.md#tc010)
- [ ] Buttons block - Settings: Remove link - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/buttons.md#tc014)
- [ ] Buttons block - Settings: Synchronize with button options - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/buttons.md#tc013)

Editor Theme-2

- [ ] Custom Gradients - Check that custom gradients load in the editor - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/editor-theme.md#tc004)
- [ ] Offline Support - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/editor-theme.md#tc005)

Color Settings-2

- [ ] Pressing color swatch - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/color-settings.md#tc004)
- [ ] Customize gradient color - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/color-settings.md#tc005)
- [ ] Custom color picker - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/color-settings.md#tc006)

**Comment by geriux** [comment]
Hey @chipsnyder 👋  I had the [same issues ](https://github.com/wordpress-mobile/WordPress-Android/pull/12588#issuecomment-669998301) from the previous release related to the custom gradients. I guess it is still the jetpack sync issue right?

**Comment by chipsnyder** [comment]
> I guess it is still the jetpack sync issue right?

👍 Yup, @enejb is looking into it still. Hopefully, we can get it resolved in the next Jetpack release.

**Comment by geriux** [comment]
> > I guess it is still the jetpack sync issue right?
> 
> 👍 Yup, @enejb is looking into it still. Hopefully, we can get it resolved in the next Jetpack release.

Sounds good! Thanks!

**Comment by geriux** [comment]
We are all good on the sanity tests ✅ The failed ones of the Color settings are for a known issue that we are working on but it's not a blocker. The failed test with the Spacer block is not a blocker either, I think we need to update that test case.

**Review by ceyhun — COMMENTED** [review]
CI fails with `Podfile.lock was changed by 'pod install'` error but I didn't get any changes locally running `rm -rf Pods && bundle exec pod install --repo-update`.
Could be something to do with `RNCMaskedView`. 
Can't approve as I'm the author but :shipit:

File: Podfile.lock [hunk 1]
```diff
@@ -573,106 +573,106 @@ SPEC REPOS:
...
   FSInteractiveMap:
     :git: https://github.com/wordpress-mobile/FSInteractiveMap.git
     :tag: 0.2.0
   Gutenberg:
-    :commit: bd76109049d2eeead78478d20d66ffd7b1567dd3
+    :commit: 6996aa39c3aaf9684bc7925fc6c362ade77a8583
     :git: http://github.com/wordpress-mobile/gutenberg-mobile/
     :submodules: true
   RNCMaskedView:
     :commit: d4ccf2bba163679c4550ce6ba0119604cd5e6379
```
Comment (ceyhun) [inline Podfile.lock:672]: 🤨

**Review by cameronvoell — APPROVED** [review]
Came to same conclusions as @ceyhun 
> CI fails with `Podfile.lock was changed by 'pod install'` error but I didn't get any changes locally running `rm -rf Pods && bundle exec pod install --repo-update`.
> Could be something to do with `RNCMaskedView`.
> Can't approve as I'm the author but :shipit:

LGTM :shipit:

**Comment by cameronvoell** [comment]
Though `rake dependencies` does not seem to always agree with d0781adedf78e3556c4a9cc4a57ea38d328e4f9d, I was able to manually verify the checksum using:

>cameronvoell$ pod ipc spec RNCMaskedView.podspec | openssl sha1
5a8ec07677aa885546a0d98da336457e2bea557f

So we'll merge for now, while continuing to investigate why this change d0781adedf78e3556c4a9cc4a57ea38d328e4f9d is not produced consistently for all developers

**Pull request merged**